### PR TITLE
feat: rename card templates 

### PR DESCRIPTION
**Pull request by david-allison**
## Issue
* Issue #6017

## How Has This Been Tested?
Screenshot 2024-02-24 at 14 51 22
Screenshot 2024-02-24 at 14 51 46

## Checklist
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/notetype/RenameNoteTypeDialog.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import android.content.Context
+import androidx.appcompat.app.AlertDialog
+import com.ichi2.anki.CollectionManager
+import com.ichi2.anki.R
+import com.ichi2.utils.input
+import com.ichi2.utils.negativeButton
+import com.ichi2.utils.positiveButton
+import com.ichi2.utils.show
+
+class RenameNoteTypeDialog {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/notetype/RenameNoteTypeDialog.kt:28]: rename

**Review by mikehardy — APPROVED** [review]
I'm okay with picking the fix and the feature as long as it passes at least a manual click through without crashing on the various cases of renaming normal, cloze, and image occlusion.

At minimum picking the first commit though yes. I'll be reading these comments before I do the pick so discussion here will be taken into consideration / inform the pick

**Comment by david-allison** [comment]
I'm happy keeping the feature for 2.18

**Review by lukstbit — APPROVED** [review]
LGTM!

**Comment by david-allison** [comment]
Only the first commit is for cherry-picking

**Pull request merged**